### ISSUE TITLE
Passing a struct from an array into a function produces invalid SPIR-V

### ISSUE THREAD

**Issue by Anteru** [KhronosGroup/glslang#1245]
The attached GLSL code produces SPIR-V which doesn't pass validation, failing with:

    error: 500: OpFunctionCall Argument <id> '67's type does not match Function <id> '8's parameter type.

Changing `#if 0` to `#if 1` makes it work flawlessly. It seems to me the problem is on line 123 of `incorrect-unpack.spvasm`, where the result gets stored into `%76` instead of `%param`, and thus the call fails. In the correct compilation, we can see:

    OpStore %param %76
    %77 = OpFunctionCall %v3float %Unpack_struct_float3_f1_f1_f11_ %param
 
While the invalid one has:

    OpStore %76 %75
    %77 = OpFunctionCall %v3float %Unpack_struct_float3_f1_f1_f11_ %param

I've reproduced this on the GLSL in the VK 1.0.68 SDK.

* [correct-unpack.spvasm](https://github.com/KhronosGroup/glslang/files/1719577/correct-unpack.spvasm.txt)
* [incorrect-unpack..spvasm](https://github.com/KhronosGroup/glslang/files/1719578/incorrect-unpack.spvasm.txt)
* [UnpackF3.comp](https://github.com/KhronosGroup/glslang/files/1719579/UnpackF3.comp.txt)

**Comment by johnkslang** [comment]
I see glslang made two types (three, actually, but these are the two at play)q for the struct:

```
  7(float3):             TypeStruct 6(float) 6(float) 6(float)
 65(float3):             TypeStruct 6(float) 6(float) 6(float)
```

This is allowed, to support different kinds of decorations, but code is needed to copy one to the other.

**Comment by johnkslang** [comment]
FYI, simplified shader that reproduces this:

```
#version 450

struct float3
{
    float x, y, z;
};

void Unpack (float3 f)
{
}

layout(set = 0, binding = 0) buffer InputBuffer
{
    float3 inputData[];
};

void main()
{
    Unpack (inputData[3]);
}
```

**Comment by LapisSea** [comment]
This is also happening but to add to that, when I do: (expanding on @johnkslang code)
```
float3 f=inputData[3];
Unpack (f)
```
in an effort to bypass this bug.
It does not give an error when creating shader module but instead of that crashes the driver (all screens go black for a moment) after a few seconds of application running with error of lost logical or physical device. (I'm on Windows 10, with a GTX 970 driver V 391.01)
I've recorded a short video demonstrating the bug: https://youtu.be/MVwGNJFzdp4

**Comment by feliwir** [comment]
Same bug here

**Comment by johnkslang** [comment]
This is fixed with d2e9829a88fc83a5a93d4cfc0423d5d9331fadc1
